packages/kit/src/core/postbuild/prerender.js

```diff
@@ -176,6 +190,31 @@ async function prerender({ hash, out, manifest_path, metadata, verbose, env }) {
 		return file;
 	}
 
+	/**
+	 * @param {string} content_type
+	 * @returns {string}
+	 */
+	function get_extension_from_content_type(content_type) {
+		// Extract the mime type (ignore charset and other parameters)
+		const mime = content_type.split(';')[0].trim().toLowerCase();
+
+		// Common mime type to extension mappings
+		/** @type {Record<string, string>} */
+		const mime_to_ext = {
+			'application/json': '.json',
+			'text/plain': '.txt',
+			'text/css': '.css',
+			'text/javascript': '.js',
+			'application/javascript': '.js',
+			'application/xml': '.xml',
+			'text/xml': '.xml',
+			'application/pdf': '.pdf',
+			'image/svg+xml': '.svg'
+		};
+
+		return mime_to_ext[mime] || '';
```

Comment:
We can probably use the `mrmime` package that's already installed to do this. So something like:
```js
import { mimes } from 'mrmime';

// TODO: `mimes` is a map of file extensions to mime types but we want it the other way around
```